sprockets/mixins/correlation/mixins.py

```diff
@@ -39,11 +42,15 @@ def __init__(self, *args, **kwargs):
         self.__correlation_id = str(uuid.uuid4())
         super(HandlerMixin, self).__init__(*args, **kwargs)
 
+    @gen.coroutine
     def prepare(self):
         # Here we want to copy an incoming Correlation-ID header if
         # one exists.  We also want to set it in the outgoing response
         # which the property setter does for us.
-        super(HandlerMixin, self).prepare()
+        maybe_future = super(HandlerMixin, self).prepare()
+        if maybe_future:
```

Comment:
You sure it doesn't make sense to guard this with ``if concurrent.is_future(maybe_future):``?